nodes/modifier_make/adaptive_polygons_mk3.py

```diff
@@ -57,6 +57,14 @@
 sqrt_3_3 = sqrt_3/3
 sqrt_3_2 = sqrt_3/2
 
+def normalize_v3(arr):
+    ''' Normalize a numpy array of 3 component vectors shape=(n,3) '''
+    lens = np_sqrt( arr[:,0]**2 + arr[:,1]**2 + arr[:,2]**2 )
+    arr[:,0] /= lens
+    arr[:,1] /= lens
+    arr[:,2] /= lens
+    return arr
```

Comment:
I would suggest
1. Change this to something like
```python
result = np.zeros(arr.shape)
norms = np.linalg.norm(arr, axis=1, keepdims=True)
nonzero = (norms > 0)[:,0]
result[nonzero] = arr[nonzero] / norms[nonzero][:,0][np.newaxis].T
```
this 1) uses vectorization more effectively, 2) takes care of division by zero.

2. Move it to `sverchok.utils.math`, because I think I saw similar functions in other modules, at least next time there will be "standard" function to use.

Reply:
Ok.
I did some performance tests
and this was 10% faster:
```
def np_normalized_v3(vecs):
    result = np.zeros(vecs.shape)
    norms = np.linalg.norm(vecs, axis=1)
    nonzero = (norms > 0)
    result[nonzero] = vecs[nonzero] / norms[nonzero][:,np.newaxis]
    return result
```
And this like 2.5x faster. Does the normalisation in-place (good for me in this case)
```
def np_normalize_v3(vecs):
    norms = np.linalg.norm(vecs, axis=1)
    nonzero = (norms > 0)
    vecs[nonzero] = vecs[nonzero] / norms[nonzero][:,np.newaxis]
```
I added both to `sverchok.utils.math`